### ISSUE TITLE
[BUG] 'cpython/int.pxd' not found: build failure with Cython master

### ISSUE THREAD

**Issue by oscarbenjamin** [cython/cython#6332]
### Describe the bug

With current release (Cython 3.0.11) all is fine but with current Cython master I am unable to build python-flint:
```console
$ spin test
Invoking `build` prior to running tests:
$ meson compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/oscar/.pyenv/shims/ninja -C /home/oscar/current/active/python-flint/build
ninja: Entering directory `/home/oscar/current/active/python-flint/build'
[1/102] Compiling Cython source /home/oscar/current/active/python-flint/src/flint/types/fmpz.pyx
FAILED: src/flint/types/fmpz.cpython-311-x86_64-linux-gnu.so.p/src/flint/types/fmpz.pyx.c 
cython -M --fast-fail -3 -X embedsignature=True -X emit_code_comments=True /home/oscar/current/active/python-flint/src/flint/types/fmpz.pyx -o src/flint/types/fmpz.cpython-311-x86_64-linux-gnu.so.p/src/flint/types/fmpz.pyx.c

Error compiling Cython file:
------------------------------------------------------------
...
from cpython.int cimport PyInt_Check, PyInt_AS_LONG
^
------------------------------------------------------------

/home/oscar/current/active/python-flint/src/flint/types/fmpz.pxd:1:0: 'cpython/int.pxd' not found
```

### Code to reproduce the behaviour:

I have not extracted a minimal reproducer yet.

**Comment by oscarbenjamin** [comment]
I'm not sure how to reproduce this in a simple example. I tried this but it seems to compile despite warnings:
```console
$ cat stuff.pyx
from cpython.int cimport PyInt_Check, PyInt_AS_LONG
$ cython stuff.pyx 
/home/oscar/.pyenv/versions/3.11.3/envs/python-flint-3.11/lib/python3.11/site-packages/Cython/Compiler/Main.py:373: FutureWarning: Cython directive 'language_level' not set, using '3' (Py3). This has changed from earlier releases! File: /home/oscar/current/active/python-flint/stuff.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)
```

**Comment by da-woods** [comment]
OK... it works on Cython 3.0 because we define

```
#if PY_MAJOR_VERSION >= 3
  // ...
  #define PyInt_Check(op)              PyLong_Check(op)
  // ...
  #define PyInt_FromLong               PyLong_FromLong
  // ...
#endif
```

To me that suggests we should restore `cpython.int` pxd file and add a warning to give a slightly softer deprecation/break. Probably not for the other Python 2 pxd files though.

**Comment by oscarbenjamin** [comment]
Okay, thanks.

So it should be possible to fix this on python-flint's side by not using cpython.int because we should use the `PyLong_*` functions instead.

**Comment by oscarbenjamin** [comment]
> I'm not sure how to reproduce this in a simple example. I tried this but it seems to compile despite warnings:

Now when I try the sample fail it fails with the same error messages:
```python
$ cython stuff.pyx 
/home/oscar/.pyenv/versions/3.11.3/envs/python-flint-3.11/lib/python3.11/site-packages/Cython/Compiler/Main.py:373: FutureWarning: Cython directive 'language_level' not set, using '3' (Py3). This has changed from earlier releases! File: /home/oscar/current/active/python-flint/stuff.pyx
  tree = Parsing.p_module(s, pxd, full_module_name)

Error compiling Cython file:
------------------------------------------------------------
...
from cpython.int cimport PyInt_Check, PyInt_AS_LONG
^
------------------------------------------------------------

stuff.pyx:1:0: 'cpython/int.pxd' not found

Error compiling Cython file:
------------------------------------------------------------
...
from cpython.int cimport PyInt_Check, PyInt_AS_LONG
^
------------------------------------------------------------

stuff.pyx:1:0: 'cpython/int/PyInt_Check.pxd' not found

Error compiling Cython file:
------------------------------------------------------------
...
from cpython.int cimport PyInt_Check, PyInt_AS_LONG
^
------------------------------------------------------------

stuff.pyx:1:0: 'cpython/int/PyInt_AS_LONG.pxd' not found
```
I have had some difficulty investigating the effect of different Cython commits because it seems that I don't always get a clean rebuild of Cython. It seems I have to do this each time or otherwise checking out a new commit does not take effect:
```python
cd python-flint
pip uninstall cython
cd ../cython
git clean -fdx  # Nuke all build files. Is this necessary?
git checkout <some commit>
cd ../python-flint
pip install ../cython  # This should use an isolated build environment...
```

**Comment by oscarbenjamin** [comment]
I don't know whether this issue is valid or not. In https://github.com/flintlib/python-flint/pull/177 I am removing all of the relevant code as it was only needed in old Py3-specific code paths that are not needed any more.

**Comment by scoder** [comment]
We *could* keep the `int.pxd` file and move the `#define` section there.

OTOH, the error message is actually correct, the file isn't there any more. And code that wants to use Cython 3.1 should not need it. It's best for everyone to let users actually update their code.

I've added an explicit comment to the changelog that the three legacy pxd files were removed in https://github.com/cython/cython/pull/5870. That should at least make it easier to find (together with this ticket), if users need a hint what they need to do.
